test_conformance/api/test_device_command_queue.cpp

```diff
@@ -0,0 +1,88 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "testBase.h"
+#include <array>
+
+using namespace std;
+
+
+int test_command_queue_helper(cl_context context, cl_device_id deviceID,
+                              cl_command_queue queue)
+{
+    cl_int error;
+    cl_command_queue check_queue;
+
+    error = clSetDefaultDeviceCommandQueue(context, deviceID, queue);
+    test_error(error, "clSetDefaultDeviceCommandQueue failed ");
+
+    error = clGetCommandQueueInfo(queue, CL_QUEUE_DEVICE_DEFAULT,
+                                  sizeof(check_queue), &check_queue, nullptr);
+    test_error(error,
+               "clGetCommandQueueInfo failed for CL_QUEUE_DEVICE_DEFAULT");
+    test_assert_error(
+        (check_queue == queue),
+        "Expected the queue to be returned as default device queue failed");
+
+    return TEST_PASS;
+}
+
+int test_set_default_device_command_queue(cl_device_id deviceID,
+                                          cl_context context,
+                                          cl_command_queue queue,
+                                          int num_elements)
+{
+    cl_int error;
+    constexpr cl_command_queue_properties PROPERTIES = CL_QUEUE_ON_DEVICE
+        | CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE | CL_QUEUE_PROFILING_ENABLE;
+    std::array<cl_queue_properties, 3> properties = {
+        CL_QUEUE_PROPERTIES, (PROPERTIES | CL_QUEUE_ON_DEVICE_DEFAULT), 0
+    };
+
+    if (get_device_cl_version(deviceID) <= Version(3, 0))
+    {
+        cl_device_device_enqueue_capabilities dseCaps = 0;
+        error = clGetDeviceInfo(deviceID, CL_DEVICE_DEVICE_ENQUEUE_CAPABILITIES,
+                                sizeof(dseCaps), &dseCaps, NULL);
+        test_error(error,
+                   "Unable to query CL_DEVICE_DEVICE_ENQUEUE_CAPABILITIES");
+
+        if (0 == (dseCaps & CL_DEVICE_QUEUE_REPLACEABLE_DEFAULT))
+            return TEST_SKIP;
+    }
+
+    clCommandQueueWrapper cmd_queue_1 = clCreateCommandQueueWithProperties(
+        context, deviceID, properties.data(), &error);
+    test_error(error, "clCreateCommandQueueWithProperties failed");
+
+    properties[1] = PROPERTIES;
+    clCommandQueueWrapper cmd_queue_2 = clCreateCommandQueueWithProperties(
+        context, deviceID, properties.data(), &error);
+    test_error(error, "clCreateCommandQueueWithProperties failed");
+
+    // cmd_queue_1
+    if (test_command_queue_helper(context, deviceID, cmd_queue_1) != 0)
+    {
+        test_fail("test_command_queue_helper failed for cmd_queue_1.\n");
+    }
+
+    // cmd_queue_2 - without CL_QUEUE_ON_DEVICE_DEFAULT
+    if (test_command_queue_helper(context, deviceID, cmd_queue_2) != 0)
+    {
+        test_fail("test_command_queue_helper failed for cmd_queue_2.\n");
+    }
+
+    return TEST_PASS;
```

Comment:
The command queues are leaked from this function and need releasing.

Reply:
I think the command queue wrapper will release the queues?

Reply:
Yes you're right. I'd missed the wrapper type.